File: database/app/src/main/java/com/google/firebase/quickstart/database/fragment/MyTopPostsFragment.java
package com.google.firebase.quickstart.database.fragment;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.Query;

public class MyTopPostsFragment extends PostListFragment {

    public MyTopPostsFragment() {}

    @Override
    public Query getQuery(DatabaseReference databaseReference) {
        // [START my_top_posts_query]
        // My top posts by number of stars
        String myUserId = getUid();
        Query myTopPostsQuery = databaseReference.child("user-posts").child(myUserId)
                .orderByChild("starCount");
        // [END my_top_posts_query]

        return myTopPostsQuery;
    }
}

File: database/app/src/main/java/com/google/firebase/quickstart/database/kotlin/fragment/MyTopPostsFragment.kt
package com.google.firebase.quickstart.database.kotlin.fragment

import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.Query

class MyTopPostsFragment : PostListFragment() {

    override fun getQuery(databaseReference: DatabaseReference): Query {
        // [START my_top_posts_query]
        // My top posts by number of stars
        val myUserId = uid
        // [END my_top_posts_query]

        return databaseReference.child("user-posts").child(myUserId)
                .orderByChild("starCount")
    }
}
